### DISCUSSION TITLE
How can I send TCP packets directly through libp2p?

### DISCUSSION THREAD

**Discussion by tristan-f-r** [libp2p/rust-libp2p#2462] [Q&A] [answered]
Hello! I'm not sure if there's a better use case for this, but I would like to relay TCP packets from a server to clients using libp2p instead of port forwarding. Is there a general method I can use to do this?

**Comment by mxinden** [comment, marked as answer]
You could build your own via `libp2p-request-response`, or you make sure of what IPFS already has to offer:

https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipfs-p2p

Let me know if this is of some help.
